src/main/java/seedu/address/model/card/Card.java

```diff
@@ -27,10 +31,14 @@
     private final String front;
     private final String back;
 
-    public Card(String front, String back) {
-        requireAllNonNull(front, back);
+    private final UniqueTagList tags;
+
+    public Card(String front, String back, Set<Tag> tags) {
+        requireAllNonNull(front, back, tags);
```

Comment:
You require that tags be non-null, yet later on you specifically pass `null` to the component.